bdd-for-c.h

```diff
@@ -637,14 +595,15 @@ void __bdd_snprintf__(char *buffer, size_t bufflen, const char *fmt, const char
 
 #define __BDD_CHECK__(condition, ...) if (!(condition))\
 {\
-    const char *message = __bdd_format__(__VA_ARGS__);\
+    char *message = __bdd_format__(__VA_ARGS__);\
     const char *fmt = __bdd_config__->use_color ?\
         (__BDD_COLOR_RED__ "Check failed:" __BDD_COLOR_RESET__ " %s" ) :\
         "Check failed: %s";\
     __bdd_config__->location = "at " __FILE__ ":" __STRING__LINE__;\
     size_t bufflen = strlen(fmt) + strlen(message) + 1;\
     __bdd_config__->error = calloc(bufflen, sizeof(char));\
     __bdd_snprintf__(__bdd_config__->error, bufflen, fmt, message);\
+    free(message);\
     return;\
 }
```

Comment:
MSVC complains here that you there are non-literal strings for format so could you please replace this with:

```c
#define __BDD_FMT_COLOR __BDD_COLOR_RED__ "Check failed:" __BDD_COLOR_RESET__ " %s"
#define __BDD_FMT_PLAIN "Check failed: %s"

#define __BDD_CHECK__(condition, ...) if (!(condition))\
{\
    char *message = __bdd_format__(__VA_ARGS__);\
    const char *fmt = __bdd_config__->use_color ? __BDD_FMT_COLOR : __BDD_FMT_PLAIN;\
    __bdd_config__->location = "at " __FILE__ ":" __STRING__LINE__;\
    size_t bufflen = strlen(fmt) + strlen(message) + 1;\
    __bdd_config__->error = calloc(bufflen, sizeof(char));\
    if (__bdd_config__->use_color) {\
      snprintf(__bdd_config__->error, bufflen, __BDD_FMT_COLOR, message);\
    } else {\
      snprintf(__bdd_config__->error, bufflen, __BDD_FMT_PLAIN, message);\
    }\
    free(message);\
    return;\
}
```

This also allows to completely get rid of `__bdd_snprintf__` wrapper

Reply:
Fixed this according to your suggestion. Added double underscore suffix to the `__BDD_FMT_X__` macros for consistency.